gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -473,22 +490,20 @@ private void initialize(Dag<JobExecutionPlan> dag)
     /**
      * Proceed the execution of each dag node based on job status.
      */
-    private void pollAndAdvanceDag()
-        throws IOException {
+    private void pollAndAdvanceDag() throws IOException, ExecutionException, InterruptedException {
       this.failedDagIdsFinishRunning.clear();
-
       Map<String, Set<DagNode<JobExecutionPlan>>> nextSubmitted = Maps.newHashMap();
       List<DagNode<JobExecutionPlan>> nodesToCleanUp = Lists.newArrayList();
+
       for (DagNode<JobExecutionPlan> node: this.jobToDag.keySet()) {
-        long pollStartTime = System.nanoTime();
+        boolean slaKilled = slaKillIfNeeded(node);
+
         JobStatus jobStatus = pollJobStatus(node);
-        Instrumented.updateTimer(this.jobStatusPolledTimer, System.nanoTime() - pollStartTime, TimeUnit.NANOSECONDS);
-        if (jobStatus == null) {
-          continue;
-        }
+
+        ExecutionStatus status = getJobExecutionStatus(slaKilled, jobStatus);
```

Comment:
Why not ExecutionStatus status = (slaKilled) ? ExecutionStatus.CANCELLED : getJobExecutionStatus(pollJobStatus(node));?

Reply:
Almost the same thing, less readable and giving less emphasis on the fact that slaKilled affects the execution status.